python/wrapper_tests/test_01_synthetic_collection.py

```diff
@@ -0,0 +1,128 @@
+    ref_score_config.type = constants.ScoreConfig_Type_TopTokens
+    config = messages.TopTokensScoreConfig()
+    config.num_tokens = 4
+    ref_score_config.config = config.SerializeToString()
+
+    master_id = lib.ArtmCreateMasterComponent(master_config)
+
+    # Initialize model
+    init_args = messages.InitializeModelArgs()
+    init_args.model_name = pwt
+    init_args.disk_path = batches_folder
+    init_args.source_type = constants.InitializeModelArgs_SourceType_Batches
+    init_args.topics_count = 10
+    lib.ArtmInitializeModel(master_id, init_args)
+
+    # Create configuration for batch processing
+    proc_args = messages.ProcessBatchesArgs()
+    proc_args.pwt_source_name = pwt
+    proc_args.nwt_target_name = nwt
+    for name in os.listdir(batches_folder):
+        proc_args.batch_filename.append(os.path.join(batches_folder, name))
+    proc_args.inner_iterations_count = 10
+
+    # Create configuration for Phi normalization
+    norm_args = messages.NormalizeModelArgs()
+    norm_args.pwt_target_name = pwt
+    norm_args.nwt_source_name = nwt
+
+    # Create config for scores retrieval
+    perplexity_args = messages.GetScoreValueArgs()
+    perplexity_args.model_name = pwt
+    perplexity_args.score_name = 'PerplexityScore'
+
+    top_tokens_args = messages.GetScoreValueArgs()
+    top_tokens_args.model_name = pwt
+    top_tokens_args.score_name = 'TopTokensScore'
+
+    for iter in xrange(10):
+        # Invoke one scan of the collection and normalize Phi
+        lib.ArtmRequestProcessBatches(master_id, proc_args)
+        lib.ArtmNormalizeModel(master_id, norm_args)    
+
+        # Retrieve and print perplexity score
+        results = lib.ArtmRequestScore(master_id, perplexity_args)
+        score_data = messages.ScoreData()
+        score_data.ParseFromString(results)
+        perplexity_score = messages.PerplexityScore()
+        perplexity_score.ParseFromString(score_data.data)
+        
+        if iter == 3:
+            assert abs(perplexity_score.value - 22.5) < 1.0
+        if iter == 8:
+            assert abs(perplexity_score.value - 21.7) < 1.0
+        
+        print "Iteration#" + str(iter) + ": Perplexity = %.3f" % perplexity_score.value
+
+    # Retrieve and print top tokens score
+    results = lib.ArtmRequestScore(master_id, top_tokens_args)
+    score_data = messages.ScoreData()
+    score_data.ParseFromString(results)
+    top_tokens_score = messages.TopTokensScore()
+    top_tokens_score.ParseFromString(score_data.data)
+
+    print 'Top tokens per topic:',
+    topic_index = -1
+    for i in xrange(top_tokens_score.num_entries):
+        if top_tokens_score.topic_index[i] != topic_index:
+            topic_index = top_tokens_score.topic_index[i]
+            print "\nTopic#" + str(topic_index + 1) + ": ",
+        assert abs(top_tokens_score.weight[i] - 0.1) < 0.15
+        print top_tokens_score.token[i] + "(%.3f) " % top_tokens_score.weight[i],
+    print '\n',
```

Comment:
Don't print incomplete lines (the lines without the '\n').
Instead, collect all the data needed to print the whole line and then print.